node/src/components/block_proposer/tests.rs

```diff
@@ -354,16 +356,16 @@ fn should_keep_track_of_unhandled_deploys() {
         !proposer.sets.finalized_deploys.contains_key(deploy2.id()),
         "should not yet contain deploy2"
     );
+    assert!(
+        proposer.contains_finalized(deploy2.id()),
+        "should recognize deploy2 as finalized"
+    );
 
-    let past_deploys = HashSet::new();
     assert!(
-        proposer.is_deploy_valid(
-            deploy2.header(),
-            test_time,
-            &proposer.deploy_config,
-            &past_deploys
-        ),
-        "deploy2 should -not- be valid"
+        deploy2
+            .header()
+            .is_valid(&proposer.deploy_config, test_time),
+        "deploy2 should be valid"
```

Comment:
Why did this assertion change? It also does not expect the `past_deploys` anymore, is that test equivalent now?

Reply:
It is, and the "not" must have been a typo, because the assertion was that it _was_ valid.
Anyway, we now check that the header `is_valid` and above that `proposer.contains_finalized`, which I believe was the original intention. Maybe @dwerner can confirm?

Reply:
The original intent of this test was to ensure that the block proposer would correctly handle the case where:

- We received a finalized deploy hash (through `.finalized_deploys`)
- We have NOT yet received the DeployType

This scenario can happen since messages from the reactor can arrive out of order.

From reading the above test case, it would appear the original intent is preserved, minus my typo.